src/status_im2/contexts/chat/photo_selector/view.cljs

```diff
@@ -0,0 +1,108 @@
+(ns status-im2.contexts.chat.photo-selector.view
+  (:require [i18n.i18n :as i18n]
+            [quo.components.safe-area :as safe-area]
+            [quo2.components.notifications.info-count :as info-count]
+            [quo2.core :as quo2]
+            [quo2.foundations.colors :as colors]
+            [react-native.core :as rn]
+            [react-native.linear-gradient :as linear-gradient]
+            [reagent.core :as reagent]
+            [status-im2.contexts.chat.photo-selector.style :as style]
+            [status-im.utils.core :as utils]
+            [utils.re-frame :as rf]))
+
+(def selected (reagent/atom []))
```

Comment:
You can avoid the global state and make it local to the top-level component, then pass it as an argument to downstream functions. This is literally a global mutable var, initialized outside the React rendering cycle, so I think it's quite obvious we should avoid this.

Even though other places in the codebase do this, IMO this is not the way to go in React, in Re-frame, and even in Clojure apps in general.

Reply:
But that would be kinda "prop drilling". There are 4 different components in that file, all of which use the `selected` state, so I think making it global in that file is better (?)

Reply:
> But that would be kinda "prop drilling". There are 4 different components in that file, all of which use the selected state, so I think making it global in that file is better (?)

But that's exactly the argument developers often use about any state in a any application ever, "let me create a global state here so that I don't have to pass things around". So I don't see this as a good enough reason.

Prop drilling is a much more severe issue IMO, where components have to pass state deep down the component hierarchy, usually across lots of different files and it causes massive maintenance issues. Here it's just 4 functions in a single namespace.

> The selected state is updated on-press on an image, not part of any lifecycle.

It really is part of a lifecycle, even though it's not triggered immediately. `selected` is a reactive atom, and changing this global state triggers a whole cascade of reprocessing in Re-frame > Reagent > React.

---

So we can all disagree about this topic of how to manage local state in components, but it's a constant source of discussion in PRs. Maybe it's about time we have some definition around this topic (not in this PR of course @OmarBasem, so I suggest we leave the discussion for another moment).

Reply:
Thanks @ilmotta for this information. I am going to try to play around with such an example and compare the renders count.